web/modules/custom/solution/src/EventSubscriber/WorkflowTransitionEventSubscriber.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Drupal\solution\EventSubscriber;
+
+use Drupal\message\Entity\Message;
+use Drupal\message\Entity\MessageTemplate;
+use Drupal\message_notify\MessageNotifier;
+use Drupal\og\Entity\OgMembership;
+use Drupal\state_machine\Event\StateChangeEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Event subscribers for Organic Groups.
+ */
+class WorkflowTransitionEventSubscriber implements EventSubscriberInterface {
+
+
+  /**
+   * The message notifier service.
+   *
+   * @var \Drupal\message_notify\MessageNotifier
+   */
+  protected $messageNotifier;
+
+  /**
+   * Constructs an WorkflowTransitionEventSubscriber object.
+   */
+  public function __construct(MessageNotifier $messageNotifier) {
+    $this->messageNotifier = $messageNotifier;
+  }
+
+  /**
+   * On workflow transition.
+   *
+   * @param \Drupal\state_machine\Event\StateChangeEvent $event
+   *   The state change event.
+   *
+   * @throws \Drupal\message_notify\Exception\MessageNotifyException
+   */
+  public function onChangeToValidated(StateChangeEvent $event) {
+    $entity = $event->getEntity();
+    $bundle = $entity->bundle();
+    if ($bundle != 'solution') {
+      return;
+    }
+    $message_template = MessageTemplate::load('workflow_transition');
+
+    $storage = \Drupal::entityManager()->getStorage('og_membership');
+
+    // Sent a message to all solution administrators.
+    $membership_query = $storage->getQuery();
+    $membership_query
+      ->condition('state', 'active')
+      ->condition('entity_id', $entity->id())
+      ->condition('roles', 'rdf_entity-solution-administrator');
```

Comment:
I will implement the solution permission. Since this is the first use case, I will set the permission within the solution module as the configuration templates regarding the messages are there as well. Later on, when we have more than one use cases, we can determine whether to leave these permissions separately or create generic ones.